h2o-algos/src/main/java/hex/anovaglm/AnovaGLMUtils.java

```diff
@@ -0,0 +1,454 @@
+    aModel._output._training_metrics = glmModel._output._training_metrics;
+    for (Key<ModelMetrics> modelMetricsKey : glmModel._output.getModelMetrics()) {
+      aModel.addModelMetrics(modelMetricsKey.get().deepCloneWithDifferentModelAndFrame(glmModel, trainingFrame));
+    }
+    aModel._output._scoring_history = copyTwoDimTable(glmModel._output._scoring_history, "glm scoring history");
+  }
+
+  /***
+   * Simple method to extract GLM Models from GLM ModelBuilders.
+   * @param glmResults: array of GLM ModelBuilders
+   * @return: array of GLMModels
+   */
+  public static GLMModel[] extractGLMModels(GLM[] glmResults) {
+    int numberModel = glmResults.length;
+    GLMModel[] models = new GLMModel[numberModel];
+    for (int index = 0; index < numberModel; index++) {
+      models[index] = glmResults[index].get();
+      Scope.track_generic(models[index]);
+    }
+    return models;
+  }
+  
+  public static void removeFromDKV(Frame[] trainingFrames, int numFrame2Delete) {
+    for (int index=0; index < numFrame2Delete; index++)
+      DKV.remove(trainingFrames[index]._key);
+  }
+
+  /***
+   * This method is used to attach the weight/offset columns if they exist and the response columns, specific 
+   * transformed columns to a training frames.
+   * 
+   * @param predNums: number of all predictor combos
+   * @param allCols: Frame containing all transformed columns
+   * @param transformedColNames: transformed predictor combo arrays containing only predictor combos for a specific
+   *                           training dataset.  Recall that models are built with one predictor combo left out.  This
+   *                           is to generate that training frame with a specific predictor combo left out.
+   * @param parms: AnovaGLMParameters
+   * @return training frame excluding a specific set of predictor combos.
+   */
+  public static Frame buildSpecificFrame(int[] predNums, Frame allCols, String[][] transformedColNames, 
+                                         AnovaGLMParameters parms) {
+    final Frame predVecs = new Frame(Key.make());
+    int numVecs = predNums.length;
+    for (int index = 0; index < numVecs; index++) {
+      int predVecNum = predNums[index];
+      predVecs.add(allCols.subframe(transformedColNames[predVecNum]));
+    }
+    if (parms._weights_column != null)
+      predVecs.add(parms._weights_column, allCols.vec(parms._weights_column));
+    if (parms._offset_column != null)
+      predVecs.add(parms._offset_column, allCols.vec(parms._offset_column));
+    predVecs.add(parms._response_column, allCols.vec(parms._response_column));
+    return predVecs;
+  }
+  
+  public static GLMParameters[] buildGLMParameters(Frame[] trainingFrames, AnovaGLMParameters parms) {
+    int numberOfModels = trainingFrames.length;
+    GLMParameters[] glmParams = new GLMParameters[numberOfModels];
+    List<String> anovaGLMOnlyList = Arrays.asList("save_transformed_framekeys", "type");
```

Comment:
this could be static private field

Reply:
I made the field final.  I want to keep the method static and cannot access class members.